lte/gateway/python/integ_tests/s1aptests/test_attach_he_policy.py

```diff
@@ -0,0 +1,232 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+import time
+import ipaddress
+import gpp_types
+import s1ap_types
+import s1ap_wrapper
+import json
+from integ_tests.s1aptests.s1ap_utils import SessionManagerUtil, HeaderEnrichmentUtils
+from lte.protos.policydb_pb2 import FlowMatch, HeaderEnrichment
+
+
+class TestAttachServiceWithMultiPdnsAndBearers(unittest.TestCase):
+    def setUp(self):
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper()
+        self._sessionManager_util = SessionManagerUtil()
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def test_attach_service_with_he_policy(self):
+        """
+        This test validates the pipelineD is pushing currect config
+        to envoy proxy.
+        """
+        utils = HeaderEnrichmentUtils()
+        utils.restart_envoy_service()
+
+        self._s1ap_wrapper.configUEDevice(1)
+        req = self._s1ap_wrapper.ue_req
+        ue_id = req.ue_id
+        # APN of the secondary PDN
+        ims = {
+            "apn_name": "ims",  # APN-name
+            "qci": 5,  # qci
+            "priority": 15,  # priority
+            "pre_cap": 0,  # preemption-capability
+            "pre_vul": 0,  # preemption-vulnerability
+            "mbr_ul": 200000000,  # MBR UL
+            "mbr_dl": 100000000,  # MBR DL
+        }
+
+        # APN list to be configured
+        apn_list = [ims]
+        imsi = "IMSI" + "".join([str(i) for i in req.imsi])
+        self._s1ap_wrapper.configAPN(
+            "IMSI" + "".join([str(i) for i in req.imsi]), apn_list
+        )
```

Comment:
do you really need a second APN for this test case?

Reply:
changed it to single APN test